[DataPipe] Fixing MapDataPipe docstrings

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72476
* #72475

To render the change in documentation, please pull down this PR and build the doc in `TorchData`.

Differential Revision: [D34078078](https://our.internmc.facebook.com/intern/diff/D34078078)

NivekT committed:
[DataPipe] Fixing MapDataPipe docstrings

[ghstack-poisoned]

ejguan reviewed (approved):
LGTM! Appreciate it.

NivekT committed:
Update on "[DataPipe] Fixing MapDataPipe docstrings"


To render the change in documentation, please pull down this PR and build the doc in `TorchData`.

[ghstack-poisoned]

NivekT commented:
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34078078).
